vnext/Microsoft.ReactNative/Utils/TransformableText.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <boost/algorithm/string.hpp>
+
+namespace react::uwp {
+struct TransformableText final {
+  enum class TextTransform { Undefined, None, Uppercase, Lowercase, Capitalize };
+
+  TextTransform m_textTransform{TextTransform::Undefined};
+  std::wstring m_originalText{};
+
+  std::wstring TransformText() {
+    switch (m_textTransform) {
+      case TextTransform::Undefined:
+        [[fallthrough]];
+      case TextTransform::None:
+        return m_originalText;
+      case TextTransform::Uppercase:
+        return boost::to_upper_copy(m_originalText);
+      case TextTransform::Lowercase:
+        return boost::to_lower_copy(m_originalText);
+      case TextTransform::Capitalize: {
+        PWSTR str = new wchar_t[m_originalText.length() + 1];
```

Comment:
Consider `auto str = std::make_unique<wchar_t[]>(originalText.length() + 1])` here ... https://en.cppreference.com/w/cpp/string/basic_string/data also suggests that you can just allocate & size the output `std::wstring` carefully, as long `LCMapStringW` correctly null-terminates the buffer exactly at `data()[size()]`.